src/box/lua/console.lua

```diff
@@ -611,7 +613,11 @@ local function local_read(self)
             if not lang or lang == 'lua' then
                 -- stop once a complete Lua statement is entered
                 if local_check_lua(buf) then
-                    break
+                    if not line:endswith(multiline_delim) then
+                        break
+                    end
+                    buf = buf:sub(1, -2)
+                    -- Continue reading.
                 end
```

Comment:
The code looks like it is only about Lua, but in fact this block of code works for SQL too. It seems that it comes from fec4463da9c232dbd8f3cdab8f57b52f2b86128f. Sometimes the language parameter is in `self` (for interactive run at least), but sometimes, I guess, it is actually in `box.session`. At least `local lang = self.language or box.session.language` works for me for the interactive run. But I don't know the code well enough to suggest it as a proper fix.

Reply:
I mean, the patch should work for SQL console and it works, but it seems it is due to the bug from 2017.

Reply:
I researched it and it turns out that `box.session.language` is always `nil` and actual language is stored just in `self.language` (is set as `storage.language` in `set_language()`). Fixed in intermediate commit.